src/pfe/file-watcher/server/src/controllers/projectsController.ts

```diff
@@ -27,6 +27,7 @@ import * as projectUtil from "../projects/projectUtil";
 import * as logHelper from "../projects/logHelper";
 import * as logger from "../utils/logger";
 import * as statusController from "./projectStatusController";
+import * as projectEventsController from "./projectEventsController";
```

Comment:
This introduces a circular dependency since projectEventsController imports projectsController as well. Maybe we should move the handling of `changedFilesMap` to a separate module that can then be imported by both.

Reply:
I don't think there is a "real" circulation here. Yes, both modules call each other but circular dependency is not just two module calling one another but also their initialization, calls and etc. I was able to verify that by kicking off our test suite which verifies and catches circular dependencies during runtime of the filewatcher module

Reply:
While the tests may not have caught any circular dependencies I'm not sure that means there aren't any problems since our test coverage is not 100%. Also, if we keep it this way then in the future it would be easier for others to introduce code that does cause problems without realizing it since they could just use the already imported module. The best way to prevent future problems would be to avoid modules importing each other.

Reply:
I agree with you. I have no problem with moving that into a separate module and that is the correct way to do it to avoid future issues. But just wanted to point out there are other places in the code where we do have circular module calls like this and it works because the circular calls are not full, meaning it does not have calls to function during initialization etc. If they do have real circular calls, they would be caught by the test not because of the coverage but because of how Node works - all modules are preloaded during runtime and if Node detects any circular dependencies on initialization, it will blow up.